app/models/encryptable.rb

```diff
@@ -17,7 +17,9 @@
 
 class Encryptable < ApplicationRecord
 
-  has_paper_trail on: [:touch, :update], ignore: [:tag, :type, :encrypted_data]
+  has_paper_trail on: [:touch, :update], ignore: [:tag, :type, :encrypted_data], dependent: :destroy
+
+  before_destroy :destroy_versions
```

Comment:
Why is that necessary? (not necessarily criticizing just trying to understand)

Reply:
Hi @Robin481 The papertrail versions are not deleted from the db by default, if the encryptable they correspond to is deleted. We discussed if it would make sense to keep the old versions because the api user may still want to know what happened to an encryptable that doesn't exist anymore. However, keeping the old encryptables may pollute the db quite heavily. What do you think?

Reply:
Yes deleting them makes sense and was also discussed in #587 .
However, `dependent: :destroy` should already do that, that is why I am asking. :grin: 
They probably implemented that already in #587 so that is why you have the `dependent: :destroy` line here already which should already cover deleting the old versions and therefore making your changes redundant.

Reply:
@Robin481 Yes, `dependent: :destroy` should do that but it doesn't. There might be something wrong with the way used active record with regards to `paper_trail`. I look into it.

Reply:
Update on this front?